src/lib/homestore_backend/hs_blob_manager.cpp

```diff
@@ -24,23 +24,24 @@ BlobManager::AsyncResult< blob_id_t > HSHomeObject::_put_blob(ShardInfo const& s
     const uint32_t needed_size = sizeof(ReplicationMessageHeader);
     auto req = repl_result_ctx< BlobManager::Result< BlobInfo > >::make(needed_size, io_align);
 
+    put_blob_ctx put_ctx;
     uint8_t* raw_ptr = req->hdr_buf_.bytes;
-    ReplicationMessageHeader* header = new (raw_ptr) ReplicationMessageHeader();
-    header->msg_type = ReplicationMessageType::PUT_BLOB_MSG;
-    header->payload_size = 0;
-    header->payload_crc = 0;
-    header->shard_id = shard.id;
-    header->pg_id = pg_id;
-    header->header_crc = header->calculate_crc();
+    put_ctx.repl_header_ = new (raw_ptr) ReplicationMessageHeader();
```

Comment:
Where is `delete`?

Reply:
we dont need a delete right? as the `ReplicationMessageHeader` was allocated from `req->hdr_buf_.bytes;` which is `    sisl::io_blob_safe` and will be release as the release of `req`